### PR TITLE
Feature 62 fleet management screen

### LINKED ISSUE #62: Fleet management screent

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #62

### CODE REVIEW

File: src/components/dialogs/delete-device-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Components properties
+ */
+interface Props {
+  deviceToDelete?: Device;
+  onClose: () => void;
+  onDelete: (device: Device) => Promise<void>;
+}
+
+/**
+ * Delete Device Dialog component
```
Comment (Jdallos) [inline src/components/dialogs/delete-device-dialog.tsx:17]: Missing @params docs

File: src/components/fleet-management/edit-device-drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+ */
+interface Props {
+  selectedDevice?: Device;
+  deviceModels: DeviceModel[];
+  onSave: (device: Device) => Promise<void>;
+  onClose: () => void;
+}
+
+/**
+ * Edit Device Drawer component
```
Comment (Jdallos) [inline src/components/fleet-management/edit-device-drawer.tsx:35]: MIssing @params docs

File: src/components/fleet-management/edit-device-drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    setTempDevice({
+      ...tempDevice,
+      [name]: value
+    });
+  };
+
+  /**
+   * Event handler for switch change
+   *
+   * @param _ event
```
Comment (Jdallos) [inline src/components/fleet-management/edit-device-drawer.tsx:57]:
```suggestion
   * @param _event
```

File: src/components/fleet-management/edit-device-drawer.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+      ...tempDevice,
+      approvalStatus: checked ? DeviceApprovalStatus.Approved : DeviceApprovalStatus.Pending
+    });
+  };
+
+  /**
+   * Renders device property field by type
+   *
+   * @param name name
+   * @param disabled disabled
```
Comment (Jdallos) [inline src/components/fleet-management/edit-device-drawer.tsx:71]: Missing @params

File: src/components/fleet-management/edit-device-drawer.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+   * @param deviceModel device model
+   */
+  const renderDeviceModelSelectBoxOptions = (deviceModel: DeviceModel) => (
+    <MenuItem key={deviceModel.id} value={deviceModel.id}>
+      {deviceModel.manufacturer} {deviceModel.model}
+    </MenuItem>
+  );
+
+  /**
+   * Renders device models select box
```
Comment (Jdallos) [inline src/components/fleet-management/edit-device-drawer.tsx:200]: Missing @param docs

File: src/components/fleet-management/fleet-management-table.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+interface Props {
+  devices: Device[];
+  loading: boolean;
+  loadDevices: () => Promise<void>;
+  setSelectedDevice: (device?: Device) => void;
+  setDeviceToDelete: (device?: Device) => void;
+}
+
+/**
+ * Fleet Management Table component
```
Comment (Jdallos) [inline src/components/fleet-management/fleet-management-table.tsx:35]: Missing @params docs

File: src/components/fleet-management/fleet-management-table.tsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+      {
+        field: "createdAt",
+        headerName: strings.fleetManagement.properties.createdAt,
+        flex: 1,
+        minWidth: 100,
+        valueGetter: ({ value }) => GenericUtils.formatDateTime(value as string)
+      },
+      {
+        field: "modifiedAt",
+        headerName: "Muokattu",
```
Comment (Jdallos) [inline src/components/fleet-management/fleet-management-table.tsx:135]: Localize?

File: src/components/fleet-management/fleet-management-table.tsx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+            label={strings.generic.delete}
+            onClick={() => setDeviceToDelete(devices.find((device) => device.id === id))}
+          />
+        ]
+      }
+    ],
+    [devices, setSelectedDevice, setDeviceToDelete]
+  );
+
+  const renderFooter = () => (
```
Comment (Jdallos) [inline src/components/fleet-management/fleet-management-table.tsx:162]: Missing Docs

File: src/components/fleet-management/fleet-management-table.tsx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+    <Stack direction="row" justifyContent="space-between">
+      <IconButton disabled={loading} onClick={loadDevices}>
+        <RefreshIcon />
+      </IconButton>
+      <GridFooter />
+    </Stack>
+  );
```
Comment (Jdallos) [inline src/components/fleet-management/fleet-management-table.tsx:169]: Add line break

**Review by Jdallos — APPROVED** [review]
Approved with a few comments

File: src/components/dialogs/delete-device-dialog.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/dialogs/delete-device-dialog.tsx:17]: We haven't documented the props passed for components.

File: src/components/fleet-management/edit-device-drawer.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/fleet-management/edit-device-drawer.tsx:35]: We haven't documented the props passed for components.

File: src/components/fleet-management/edit-device-drawer.tsx (hunk 3 above)
Comment (villejuutila) [reply src/components/fleet-management/edit-device-drawer.tsx:57]: Resolved

File: src/components/fleet-management/edit-device-drawer.tsx (hunk 4 above)
Comment (villejuutila) [reply src/components/fleet-management/edit-device-drawer.tsx:71]: Resolved.

File: src/components/fleet-management/edit-device-drawer.tsx (hunk 5 above)
Comment (villejuutila) [reply src/components/fleet-management/edit-device-drawer.tsx:200]: Resolved.

File: src/components/fleet-management/fleet-management-table.tsx (hunk 6 above)
Comment (villejuutila) [reply src/components/fleet-management/fleet-management-table.tsx:35]: Resolved.

File: src/components/fleet-management/fleet-management-table.tsx (hunk 8 above)
Comment (villejuutila) [reply src/components/fleet-management/fleet-management-table.tsx:162]: Resolved

File: src/components/fleet-management/fleet-management-table.tsx (hunk 9 above)
Comment (villejuutila) [reply src/components/fleet-management/fleet-management-table.tsx:169]: Resolved

File: src/components/fleet-management/fleet-management-table.tsx (hunk 7 above)
Comment (villejuutila) [reply src/components/fleet-management/fleet-management-table.tsx:135]: Resolved

File: src/components/screens/fleet-management-screen.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+      const foundDevices = await devicesApi.listDevices({});
+      setDevices(foundDevices);
+    } catch (e: any) {
+      setError(e);
+    }
+    setLoading(false);
+  };
+
+  /**
+   * Handler for save device button click
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/components/screens/fleet-management-screen.tsx:52]: Resolved.

File: src/components/screens/fleet-management-screen.tsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+      );
+      setSelectedDevice(undefined);
+    } catch (e: any) {
+      setError(e);
+    }
+    setLoading(false);
+  };
+
+  /**
+   * Handler for delete device button click
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/components/screens/fleet-management-screen.tsx:71]: Resolved.

File: src/components/screens/fleet-management-screen.tsx [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+interface Props {
+  history: History;
+  keycloak: KeycloakInstance;
+  accessToken: AccessToken;
+  exhibitions: Exhibition[];
+  deviceModels: DeviceModel[];
+}
+
+/**
+ * Fleet Management Screen component
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/components/screens/fleet-management-screen.tsx:27]: We haven't documented the props passed for components.

**Pull request merged**